feat: add for-else tag (#1643)
python-discord/bot: bot/resources/tags/for-else.md

**for-else**

> In Python it's possible to attach an `else` clause to a for loop. The code under the `else` block will be run if the `for` block is not broken out of, by either a `break`, `return`, or `raise` statement.

Reviewer: Listing `return` or `raise` here seem odd since they are non-specific to loops, they shift the flow of execution in a different manner.

I'd rather we document it similar to [this guide](https://docs.python.org/3/tutorial/controlflow.html#break-and-continue-statements-and-else-clauses-on-loops) and write it something like this:

```suggestion
In Python it's possible to attach an `else` clause to a for loop. The code under the `else` block will be run when the iterable is exhausted (there are no more items to iterate over). Code within the else block will **not** run if the loop is broken out using `break`.
```

Here's an example of its usage:
```py

> numbers = [1, 3, 5, 7, 8, 9, 11]

Reviewer: Let's demo the actual functionality here
```suggestion
numbers = [1, 3, 5, 7, 9, 11]
```

for number in numbers:
    if number % 2 == 0:

>         print("Found an even number:", number)

Reviewer: F-strings.
```suggestion
        print(f"Found an even number: {number}")
```

>         break

Reviewer: To make the output a bit more visually interesting, let's add some prints for each iteration.
```suggestion
        break
    print(f"{number} is odd.")
```
